### ISSUE TITLE
Postgresql 13. plv8 build.

### ISSUE THREAD

**Issue by aam1go** [plv8/plv8js#415]
How to install plv8 for Postgresql 13? I didnt found any solution. Should i build manually from sources and importing extension? how to resolve this question?

**Comment by JerrySievert** [comment]
yes, you need to build from source.

see: https://plv8.github.io/#building

**Comment by aam1go** [comment]
> yes, you need to build from source.
> 
> see: https://plv8.github.io/#building

Thank you. I did it. Installed python 2.7, pkg-config.

**Comment by aam1go** [comment]
Ty, installed postgresql-13 dev. python. pkg.

**Comment by aam1go** [comment]
> yes, you need to build from source.
> 
> see: https://plv8.github.io/#building

Ty, installed postgresql-13 dev. python. pkg.

But compilation fail when:

g++ -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -shared -o plv8-2.3.15.so plv8.o plv8_type.o plv8_func.o plv8_param.o coffee-script.o livescript.o -L/usr/lib/x86_64-linux-gnu -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -L/usr/lib/llvm-6.0/lib -Wl,--as-needed -Lbuild/v8/out.gn/x64.release/obj -Lbuild/v8/out.gn/x64.release/obj/third_party/icu -lv8_base -lv8_snapshot -lv8_libplatform -lv8_libbase -lv8_libsampler -lv8_libplatform -Lbuild/v8/out.gn/x64.release/obj -lrt -std=c++11 -lc++
**/usr/bin/ld: cannot find -lc++
collect2: error: ld returned 1 exit status
/usr/lib/postgresql/13/lib/pgxs/src/makefiles/../../src/Makefile.shlib:293: recipe for target 'plv8-2.3.15.so' failed
make: *** [plv8-2.3.15.so] Error 1**

/usr/bin/ld: cannot find -lc++

**Comment by aam1go** [comment]
Solved it 
$ apt install libc++-dev

$ apt install libc++abi-dev

**Comment by GradedJestRisk** [comment]
I tried @aam1go solution in docker but it failed on `plv8.cc:224:18: error: ‘oid_hash’ was not declared in this scope`
Any idea ? Dockerfile is here https://github.com/clkao/docker-postgres-plv8/pull/37/files

```
g++ -Wall -O2 -std=c++11 -fno-rtti  -Ibuild/v8/include -Ibuild/v8 -std=c++11 -I. -I./ -I/usr/include/postgresql/13/server -I/usr/include/postgresql/internal  -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -fPIC -c -o plv8.o plv8.cc
plv8.cc: In function ‘void _PG_init()’:
plv8.cc:224:18: error: ‘oid_hash’ was not declared in this scope
  hash_ctl.hash = oid_hash;
                  ^~~~~~~~
plv8.cc:224:18: note: suggested alternative: ‘uuid_hash’
  hash_ctl.hash = oid_hash;
                  ^~~~~~~~
                  uuid_hash
plv8.cc: In function ‘Datum common_pl_call_validator(FunctionCallInfo, Dialect)’:
plv8.cc:902:25: error: ‘OPAQUEOID’ was not declared in this scope
    (proc->prorettype == OPAQUEOID && proc->pronargs == 0))
                         ^~~~~~~~~
plv8.cc:902:25: note: suggested alternative: ‘OPEROID’
    (proc->prorettype == OPAQUEOID && proc->pronargs == 0))
                         ^~~~~~~~~
                         OPEROID
make: *** [Makefile.shared:90: plv8.o] Error 1
```

**Comment by JerrySievert** [comment]
https://github.com/clkao/docker-postgres-plv8/pull/37/files#diff-f91eee8c692cc1abd9981e037747520381230af3902c37322a290e4df2e8e5b2R5

postgres 13 support became available in plv8 2.3.15, this is referencing 2.3.13

**Comment by GradedJestRisk** [comment]
:pray: Jerry. I didn't know about that, but at the time already tried in a commit I didn't pushed. 
I've just pushed it so there is no confusion about the root cause.
